allennlp/nn/initializers.py

```diff
@@ -61,6 +63,49 @@ def from_params(cls, params: Params):
             return cls.by_name(choice).from_params(params)
 
 
+def uniform_unit_scaling(tensor: torch.Tensor, nonlinearity: str = "linear"):
+    """
+    An initaliser which preserves output variance for approximately gaussian
+    distributed inputs. This boils down to initialising layers using a uniform
+    distribution in the range ``(-sqrt(3/dim[0]) * scale, sqrt(3 / dim[0]) * scale)``, where
+    ``dim[0]`` is equal to the input dimension of the parameter and the ``scale``
+    is a constant scaling factor which depends on the non-linearity used.
+
+    See `Random Walk Initialisation for Training Very Deep Feedforward Networks
+    <https://www.semanticscholar.org/paper/Random-Walk-Initialization-for-Training-Very-Deep-Sussillo-Abbott/be9728a0728b6acf7a485225b1e41592176eda0b>`_
+    for more information.
+
+    Parameters
+    ----------
+    tensor : ``torch.Tensor``, required.
+        The tensor to initialise.
+    nonlinearity : ``str``, optional (default = "linear")
+        The non-linearity which is performed after the projection that this
+        tensor is involved in. This must be the name of a function contained
+        in the ``torch.nn.functional`` package.
+
+    Returns
+    -------
+    The initialised tensor.
+    """
+    if isinstance(tensor, Variable):
+        uniform_unit_scaling(tensor.data, nonlinearity)
+        return tensor
+
+    size = 1.
+    # Estimate the input size. This won't work perfectly,
```

Comment:
You're using all of the dimensions up to `-1` because the last dimension is probably the output dimension, right?  Can you be more explicit about that in the comment?